src/sentry/api/endpoints/project_group_index.py

```diff
@@ -562,13 +563,30 @@ def put(self, request, project):
         if result.get('isSubscribed') in (True, False):
             is_subscribed = result['isSubscribed']
             for group in group_list:
+                # NOTE: Subscribing without an initiating event (assignment,
+                # commenting, etc.) clears out the previous subscription reason
+                # to avoid showing confusing messaging as a result of this
+                # action. It'd be jarring to go directly from "you are not
+                # subscribed" to "you were subscribed due since you were
+                # assigned" just by clicking the "subscribe" button (and you
+                # may no longer be assigned to the issue anyway.)
                 GroupSubscription.objects.create_or_update(
                     user=acting_user,
                     group=group,
                     project=project,
-                    values={'is_active': is_subscribed},
+                    values={
+                        'is_active': is_subscribed,
+                        'reason': GroupSubscriptionReason.unknown,
```

Comment:
But empty string is falsey. :) I don't have a strong opinion either way. I imagine we just have a ton of rows currently that are empty strings.